cmd/helm/lint_test.go

```diff
@@ -27,7 +27,7 @@ func TestLintCmdWithSubchartsFlag(t *testing.T) {
 		name:      "lint good chart with bad subcharts",
 		cmd:       fmt.Sprintf("lint %s", testChart),
 		golden:    "output/lint-chart-with-bad-subcharts.txt",
-		wantError: false,
```

Comment:
The desired behavior for this changed in Helm 3, but this test was never updated. In Helm 2, we had to retain backward compatibility to Helm 2.0.0, which did not have a requirements.yaml, and which did not impose restrictions on the formatting of subcharts. But in Helm 3, we added the requirement that a subchart be present in the Chart.yaml, and that it be valid and well-formed. (The purpose of this issue is to update the linter to reflect Helm's actual requirements/assumptions in code)